src/operators.jl

```diff
@@ -277,7 +277,8 @@ Base.sum(j::JuMPDict)  = sum(values(j.tupledict))
 Base.sum(j::JuMPArray{Variable}) = AffExpr(vec(j.innerArray), ones(length(j.innerArray)), 0.0)
 Base.sum(j::JuMPDict{Variable})  = AffExpr(collect(values(j.tupledict)), ones(length(j.tupledict)), 0.0)
 Base.sum(j::Array{Variable}) = AffExpr(vec(j), ones(length(j)), 0.0)
-function Base.sum{T<:GenericAffExpr}(affs::Array{T})
+Base.sum(j::AbstractArray{Variable}) = sum([j...]) # to handle non-one-indexed arrays.
```

Comment:
`collect(::OffsetArray)` returns a copy of the `OffsetArray` (with the same indices). I found it surprisingly hard to convert from an `OffsetArray` to a regular `Array`. Maybe I'm missing something.